lib/spack/spack/spec.py

```diff
@@ -426,7 +447,56 @@ def __init__(self, spec_like, *dep_like, **kwargs):
         # Spec(a, b) will copy a but just add b as a dep.
         for dep in dep_like:
             spec = dep if isinstance(dep, Spec) else Spec(dep)
-            self._add_dependency(spec)
+            self._add_dependency(spec, (spack.DEP_BUILD, spack.DEP_INSTALL))
+
+
+    def get_dependency(self, name):
```

Comment:
Changed it to return the `DependencySpec`. I think this might be a remnant from before I added the class.